guide/additional-info/async-await.md

```diff
@@ -0,0 +1,158 @@
+
+## How to implement async/await
+
+After knowing how Promises works and what they are for, let's look at an example in which we handle multiple Promises. Let's say you wanted to react to a message in a certain order with letters (regional indicators). For this example, I will take the basic template for a discord.js bot with some es6 adjustments.
+
+```js
+const Discord = require('discord.js');
+const client = new Discord.Client();
+
+const prefix = '?';
+
+client.on('ready', () => {
+	console.log('I am ready!');
+});
+
+client.on('message', message => {
+	if (message.content === `${prefix}react`) {
+		//code inside here
+	}
+});
+
+client.login('tokeninhere');
+```
+
+So now we need to put the code in. If you don't know how node.js asynchronous execution works, you would probably try something like this:
+
+```js
+client.on('message', message => {
+	if (message.content === `${prefix}react`) {
+		message.react('🇦');
+		message.react('🇧');
+		message.react('🇨');
+	}
+});
+```
+
+But since all of these react methods are started at the same time, it would just be a race which server request finished first, so there would be no guarantee that it would react in the order you wanted it to. In order to make sure it reacts in order (a, b, c), we need to use the .then() callback from the Promises that these methods return. As a result the code we want would mostly look like this:
+
+```js
+client.on('message', message => {
+	if (message.content === `${prefix}react`) {
+		message.react('🇦').then(() => {
+			message.react('🇧').then(() => {
+				message.react('🇨').catch(error => {
+				//handle failure of the third react
+			})
+		}).catch(error => {
+			//handle failure of the second react
+		})
+	}).catch(error => {
+		//handle failure of the first react
+	})
+	}
+});
+```
+
+This code looks really messy and has redundant catch methods, so let's look how the same code would look with async/await.
+
+```js
+client.on('message', async message => {
+	if (message.content === `${prefix}react`) {
+		await message.react('🇦');
+		await message.react('🇧');
+		await message.react('🇨');
+	}
+});
+```
+
+That would mostly be the same code with async/await (note that you can only use the await keyword inside an async function!), but how do we catch Promise rejections now since we won't use .catch() anymore? That is also a useful feature with async/await; the error will be thrown if you await it so you can just wrap the awaited Promises inside a try/catch and you are 👌. 
+
+```js
+client.on('message', async message => {
+	if (message.content === `${prefix}react`) {
+		try{
+			await message.react('🇦');
+			await message.react('🇧');
+			await message.react('🇨');
+		}catch(error) {
```

Comment:
Needs spaces after the `try` braces, and `catch(error)` needs to go on its own line, for brace-style consistency with the rest of the guide.